tools/Environments/DevHome.Environments/Views/LandingPage.xaml

```diff
@@ -106,66 +113,86 @@
-                            <ScaleTransform ScaleX="-1"/>
-                        </FontIcon.RenderTransform>
-                    </FontIcon>
-                    <TextBlock x:Uid="SyncButtonTextBlock" Margin="-5 0 0 0"/>
-                </StackPanel>
-            </Button>
         </Grid>
 
-        <!-- Search and sort -->
-        <StackPanel Grid.Row="1" Orientation="Horizontal"
-                    MaxWidth="{ThemeResource MaxPageContentWidth}" Margin="{ThemeResource HeaderMargin}">
-            <!-- Search field -->
-            <AutoSuggestBox x:Uid="SearchTextBox" x:Name="SearchTextBox" MinWidth="350" MinHeight="35" QueryIcon="Find">
-                <i:Interaction.Behaviors>
-                    <ic:EventTriggerBehavior EventName="TextChanged">
-                        <ic:InvokeCommandAction
+        <Grid Grid.Row="1">
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="10*" />
+                <ColumnDefinition Width="*" />
+                <ColumnDefinition Width="auto" />
+            </Grid.ColumnDefinitions>
+                <!-- Search field -->
+                <AutoSuggestBox Grid.Column="0" x:Uid="SearchTextBox" x:Name="SearchTextBox" Width="250" Margin="17 0 0 0" HorizontalAlignment="Left">
+                    <i:Interaction.Behaviors>
+                        <ic:EventTriggerBehavior EventName="TextChanged">
+                            <ic:InvokeCommandAction
                             Command="{x:Bind ViewModel.SearchHandlerCommand, Mode=OneWay}"
                             CommandParameter="{Binding Text, ElementName=SearchTextBox}"/>
-                    </ic:EventTriggerBehavior>
-                </i:Interaction.Behaviors>
-            </AutoSuggestBox>
-            <!-- Sort field -->
-            <TextBlock x:Uid="SortByTextBlock" VerticalAlignment="Center" Margin="50 0 10 0"/>
-            <ComboBox x:Uid="SortSelectionComboBox" x:Name="SortSelectionComboBox" Margin="0 3 0 0" MinHeight="35">
-                <ComboBoxItem x:Uid="SortByName"/>
-                <ComboBoxItem x:Uid="SortByAltName"/>
-                <i:Interaction.Behaviors>
-                    <ic:EventTriggerBehavior EventName="SelectionChanged">
-                        <ic:InvokeCommandAction
-                            Command="{x:Bind ViewModel.SortHandlerCommand}"
-                            CommandParameter="{x:Bind ((ComboBoxItem)SortSelectionComboBox.SelectedItem).Content, Mode=OneWay}" />
-                    </ic:EventTriggerBehavior>
-                </i:Interaction.Behaviors>
-            </ComboBox>
+                        </ic:EventTriggerBehavior>
+                    </i:Interaction.Behaviors>
+                </AutoSuggestBox>
+                <StackPanel Grid.Column="2" Orientation="Horizontal" HorizontalAlignment="Right">
+                    <!-- Provider field -->
+                    <TextBlock x:Uid="ProviderTextBlock" VerticalAlignment="Center" Margin="50 0 5 0"/>

```

Comment:
Looks like many of your `margins` ae to apply spacing between items.  Consider using the [spacing property](https://learn.microsoft.com/uwp/api/windows.ui.xaml.controls.stackpanel.spacing?view=winrt-22621)

Reply:
But that would only give me fixed spacing for all the elements.